app/routes/goal_routes.py

```diff
@@ -1 +1,108 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from ..models.goal import Goal
+from ..models.task import Task
+from ..db import db
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+def validate_goal(goal_id):
+    try:
+        gid = int(goal_id)
+    except ValueError:
+        invalid = {"message": f"Goal id '{goal_id}' is invalid. Must be an integer."}
+        abort(make_response(invalid, 400))
+
+    goal = db.session.get(Goal, gid)
+    if not goal:
+        not_found = {"message": f"Goal with id '{goal_id}' not found."}
+        abort(make_response(not_found, 404))
+
+    return goal
+
+@goals_bp.post("")
+def create_goal():
+    request_body = request.get_json() or {}
+    if "title" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+
+    new_goal = Goal.from_dict(request_body)
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+@goals_bp.get("")
+def get_all_goals():
+    query = db.select(Goal).order_by(Goal.id)
+    goals = db.session.scalars(query)
+
+    goals_response = [g.to_dict() for g in goals]
```

Comment:
Prefer `g` to be more descriptively named, like `goal`. It doesn't take too much extra time to write out "goal" and makes the code easier to read so it's worth writing the entire word out for clarity.

```suggestion
    goals_response = [goal.to_dict() for goal in goals]
```